File: src/DBHandler/Interfaces/IDBWrapper.cs
using System;
using System.Data;

namespace DBHandler.Interfaces
{
    /// <summary>
    /// Author : Faris Jaafar
    /// Created Date : FEB 2019
    /// UPDATED : -
    /// VERSION : 1.0
    /// </summary>
    public interface IDBWrapper : IDisposable
    {
        IDbConnection CreateConnection();
        void CloseConnection();
        IDbCommand CreateCommand(string commandText, CommandType commandType, IDbConnection connection);
        IDataAdapter CreateAdapter();
        IDbDataParameter CreateParameter();
        ConnectionState CurrentConnectionState { get; }
        IDbTransaction Begin();
        void Commit();
        void Rollback();
    }
}

File: src/DBHandlerVB/Interfaces/IDBWrapper.vb
Namespace DBHandlerVB.Interfaces
    Public Interface IDBWrapper
        Inherits IDisposable
        Function CreateConnection() As IDbConnection
        Sub CloseConnection()
        Function CreateCommand(ByVal commandText As String, ByVal commandType As CommandType, ByVal connection As IDbConnection) As IDbCommand
        Function CreateAdapter() As IDataAdapter
        Function CreateParameter() As IDbDataParameter
        ReadOnly Property CurrentConnectionState As ConnectionState
        Function Begin() As IDbTransaction
        Sub Commit()
        Sub Rollback()
    End Interface
End Namespace